### DISCUSSION TITLE
Rendering vue components with v-html

### DISCUSSION THREAD

**Discussion by BobidaHombre** [nuxt/nuxt#19893] [Questions]
Hey. I got some problem to solve. I have rich text editor (tiptap-vue3), where I can place custom vue components. The result is html with vue component inside. Example:

```
<p> Lorem Ipsum</p>
<example-component />
```

The problem is that vue components included into string don't want to render. I tried to declare it as custom component, but because of shadow DOM this approach doesn't fit me. I would like to be able to use my global styling (including Tailwind) and all Nuxt features. Do You know any approach that can help me? It can be only client-side.

**Comment by wobsoriano** [comment]
Have you tried wrapping it in a `<ClientOnly></ClientOnly>` component?

**Comment by huang-julien** [comment]
Hi v-html doesn't render any component since it directly render the html using innerHTML.
Check https://vuejs.org/api/built-in-directives.html#v-html

**Comment by ndragun92** [comment]
Maybe you will have to return it through a render function or parse HTML and inject custom components so Vue can recognize them

**Comment by huang-julien** [comment]
you can also use [nuxt-runtime-compiler](https://www.npmjs.com/package/nuxt-runtime-compiler) a  PR is currently in progress to integrate it directly with Nuxt... I just have to update it

**Comment by ddavid93** [comment]
What about this?

[https://github.com/ddavid93/vue-html-renderer](https://github.com/ddavid93/vue-html-renderer)
